catroid/src/main/java/org/catrobat/catroid/formulaeditor/FormulaElement.java

```diff
@@ -739,53 +739,145 @@ private Object interpretOperator(Operators operator, Sprite sprite) {
 
 			switch (operator) {
 				case PLUS:
+					if (leftObject instanceof Boolean || rightObject instanceof Boolean) {
+						return Double.NaN;
+					}
```

Comment:
@Deepak-Avadhya, thank you very much for contributing! :smiley: 

For backward and Scratch compatibility, calculations with numbers and booleans should be possible. `1 == true`, `0 == false`. We can't change that.

It's not well defined how to proceed in this ticket. I would try to find out how to detect such expressions like `1 < 2 < 3` first. Also if we are able to detect more complex forms and if there are limits. Then discuss the research with the Team by posting a comment in the Jira ticket.

Unfortunately I have to close your ticket. :cry: